lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -269,6 +269,19 @@ void FrontendInputsAndOutputs::addPrimaryInputFile(StringRef file,
   addInput(InputFile(file, true, buffer));
 }
 
+void FrontendInputsAndOutputs::overrideBufferForInput(
```

Comment:
Nicely encapsulated function & nice comment. Is there a comment somewhere explaining completion buffers in general? Could there be a pointer to that here, too?

Reply:
I don't understand them well enough to write that comment. :-( @benlangmuir